src/main/java/seedu/address/model/person/GithubUsername.java

```diff
@@ -0,0 +1,68 @@
+package seedu.address.model.person;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.AppUtil.checkArgument;
+
+/**
+ * Represents a Person's location in the address book.
+ * Guarantees: immutable; is valid as declared in {@link #isValidUsername(String)} (String)}
+ */
+public class GithubUsername {
+
+    public static final String MESSAGE_CONSTRAINTS = "Github username must follow the following rules: \n"
+            + "Github username cannot have multiple consecutive hyphens.\n"
+            + "Github username cannot begin or end with a hyphen.\n"
+            + "Maximum is 39 characters.";
+
+    public static final String DEFAULT_USERNAME = "";
+
+    /*
+     * Github username may only contain alphanumeric characters or hyphens.
+     * Github username cannot have multiple consecutive hyphens.
+     * Github username cannot begin or end with a hyphen.
+     * Maximum is 39 characters.
+     */
+    public static final String VALIDATION_REGEX = "^(?i)[a-z\\d](?:[a-z\\d]|-(?=[a-z\\d])){0,38}$";
+
+    public final String value;
+
+    /**
+     * Constructs a Location.
```

Comment:
Missed a change here. Could it be "Constructs a GithubUsername"?

Reply:
Opps. Thanks for pointing out.